# Establishing Comparability between the Shona Symptom Questionnaire

(SSQ-14) and the Patient Health Questionnaire (PHQ-9) as Depression Outcome Measures among Adolescents and Young Adults in South Africa

**Statistical Consideration and Analysis Report (SCAR)**

CO-LUMINATE DSU Team (Africa Health Research Institute (AHRI))

In [ ]:
## This chunk handles all package installations and loading for the factor analysis.
require(here)

Loading required package: here

here() starts at /Users/Marothi.Letsoalo/Library/CloudStorage/OneDrive-SharedLibraries-AHRI/Data Science Unit - Biostatistics/team_members/Marothi/projects/CO-LUMINATE/OBJ03-Psychometric

- The lockfile is already up to date.

Loading required package: conflicted

Loading required package: tidyverse

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.6
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.1     ✔ tibble    3.3.0
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.2.0     

Loading required package: labelled

Loading required package: reshape2

Loading required package: gtsummary

Loading required package: gt

Loading required package: gtreg

Loading required package: gtExtras

Loading required package: scales

Loading required package: ggsci

Loading required package: patchwork

Loading required package: BiocManager

Loading required package: htmltools

Loading required package: ComplexUpset

Loading required package: ggstatsplot

You can cite this package as:
     Patil, I. (2021). Visualizations with statistical details: The 'ggstatsplot' approach.
     Journal of Open Source Software, 6(61), 3167, doi:10.21105/joss.03167

Loading required package: psych

Loading required package: lavaan

This is lavaan 0.6-20
lavaan is FREE software! Please report any bugs.

Loading required package: semTools

###############################################################################

This is semTools 0.5-7

All users of R (or SEM) are invited to submit functions or ideas for functions.

###############################################################################

Loading required package: pROC

Type 'citation("pROC")' for a citation.

Loading required package: parallel

Loading required package: doParallel

Loading required package: foreach


Attaching package: 'foreach'

The following objects are masked from 'package:purrr':

    accumulate, when

Loading required package: iterators

- The lockfile is already up to date.

In [ ]:
## This chunk sets global options for figures and tables
theme_gtsummary_compact()

Setting theme "Compact"

Setting theme "The Lancet"

In [ ]:
# Create directory for statistical analysis
# Define the root directory name
root_dir <- here("statistical_analysis")

# Define the full list of directories to create
# Use file.path for OS-independent path construction
dirs_to_create <- c(
  file.path(root_dir, "data", "metadata"),
  file.path(root_dir, "data", "adam"),
  file.path(root_dir, "programs/analysis"),
  file.path(root_dir, "programs/utils"),
  file.path(root_dir, "models"),
  file.path(root_dir, "output", "table"),
  file.path(root_dir, "output", "figures"),
  file.path(root_dir, "output", "listing"),
  file.path(root_dir, "output", "objects")
)

# Create all directories recursively
# The 'showWarnings = FALSE' argument prevents warnings if the directories already exist
invisible(sapply(dirs_to_create, dir.create, recursive = TRUE, showWarnings = FALSE))

In [ ]:
## Load datasets
load(file=here("../../_private_use/data_management/data/co-luminate/adam/dt_psychometric.RData"))
load(file=here("../../_private_use/data_management/data/co-luminate/adam/dt_dreams_multi_ssq.RData"))

#load(file=here("../../_private_use/data_management/data/co-luminate/adam/dt_disposition.RData"))
#load(file=here("../../_private_use/data_management/data/co-luminate/adam/dt_concordance.RData"))

In [ ]:
## Assign main Dataset
data_name <- "dt_psychometric"
dt_main   <- get(data_name)

In [ ]:
# Source the Helper functions for Psychometric Monte Carlo EFA Pipeline
func_source_here_files(path_components = c("statistical_analysis", "programs", "utils", "efa_stability"))

Sourcing 6 file(s) from: /Users/Marothi.Letsoalo/Library/CloudStorage/OneDrive-SharedLibraries-AHRI/Data Science Unit - Biostatistics/team_members/Marothi/projects/CO-LUMINATE/OBJ03-Psychometric/statistical_analysis/programs/utils/efa_stability

 → 01_data_split_and_suitability.R

 → 02_evaluate_all_splits.R

 → 03_retention_loadings_eigen.R

 → 04_cooccurrence_consensus.R

 → 05_run_pipeline_efa_master.R

 → other_helpers.R

  STARTING ANALYSIS PIPELINE: Isisekelo Sempilo

Generated 30000 sets of split indices for Isisekelo Sempilo (includes buffer)


  Processing Item Set: SSQ-10 (Theoretical Depressive Symptoms)...

    Found existing saved results.

    Loading existing file (non-interactive mode).

    Loaded results from: /Users/Marothi.Letsoalo/Library/CloudStorage/OneDrive-SharedLibraries-AHRI/Data Science Unit - Biostatistics/team_members/Marothi/projects/CO-LUMINATE/OBJ03-Psychometric/statistical_analysis/output/objects/mc_efa/mc_10000_Isisekelo_Sempilo_SSQ-10_(Theoretical_Depressive_Symptoms)_results.rds


  Processing Item Set: PHQ-09 (Original)...

    Found existing saved results.

    Loading existing file (non-interactive mode).

    Loaded results from: /Users/Marothi.Letsoalo/Library/CloudStorage/OneDrive-SharedLibraries-AHRI/Data Science Unit - Biostatistics/team_members/Marothi/projects/CO-LUMINATE/OBJ03-Psychometric/statistical_analysis/output/objects/mc_efa/mc_10000_Isisekelo_Sempilo_PHQ-09_(Original)_results.rds


  Processing Item Set: SSQ-14 (Original)...

    Found existing saved results.

    Loading existing file (non-interactive mode).

    Loaded results from: /Users/Marothi.Letsoalo/Library/CloudStorage/OneDrive-SharedLibraries-AHRI/Data Science Unit - Biostatistics/team_members/Marothi/projects/CO-LUMINATE/OBJ03-Psychometric/statistical_analysis/output/objects/mc_efa/mc_10000_Isisekelo_Sempilo_SSQ-14_(Original)_results.rds


  Processing Item Set: SSQ-08 (Dixon Chibanda et.al)...

    Found existing saved results.

    Loading existing file (non-interactive mode).

    Loaded results from: /Users/Marothi.Letsoalo/Library/CloudStorage/OneDrive-SharedLibraries-AHRI/Data Science Unit - Biostatistics/team_members/Marothi/projects/CO-LUMINATE/OBJ03-Psychometric/statistical_analysis/output/objects/mc_efa/mc_10000_Isisekelo_Sempilo_SSQ-08_(Dixon_Chibanda_et.al)_results.rds

In [ ]:
output_dir <- here("statistical_analysis/output/objects/mc_efa")
if (!dir.exists(output_dir)) {
  dir.create(output_dir, recursive = TRUE)
}

saveRDS(
  mc_efa_results,
  file=file.path(output_dir,
                 paste0("mc_efa_",
                        n_boot_iterations,
                        "_samples_Isisekelo_Sempilo.rds"))
)
mc_efa_results<-readRDS(
  file=file.path(output_dir,
                 paste0("mc_efa_",
                        n_boot_iterations,
                        "_samples_Isisekelo_Sempilo.rds"))
)

In [ ]:
############################################################
# SSQ / PHQ Monte Carlo CFA Validation
# 07_batch.R - Batch runner for multiple configurations
############################################################

# Source the Helper functions for Psychometric Monte Carlo EFA Pipeline
func_source_here_files(path_components = c("statistical_analysis", "programs", "utils", "cfa_roc"))

Sourcing 7 file(s) from: /Users/Marothi.Letsoalo/Library/CloudStorage/OneDrive-SharedLibraries-AHRI/Data Science Unit - Biostatistics/team_members/Marothi/projects/CO-LUMINATE/OBJ03-Psychometric/statistical_analysis/programs/utils/cfa_roc

 → 00_setup.R

 → 01_efa_consensus.R

 → 02_invariance_roc_helpers.R

 → 03_mc_loop.R

 → 04_summary.R

 → 05_plots.R

 → 06_master.R

Data loaded successfully. Rows: 843 Cols: 44

Target Set: SSQ-10 (Theoretical Depressive Symptoms)

Items: SSQ01, SSQ02, SSQ03, SSQ08, SSQ09, SSQ10, SSQ11, SSQ12, SSQ13, SSQ14

Initial Consensus: Kaiser suggested 1 Factors

Warning in select_mc_seeds(valid_runs = valid_runs, winning_sig = winning_sig,
: No seeds matched the winning signature; returning empty vector.

Re-using existing Monte Carlo results from disk.

ℹ In argument: `Mean_Cutoff = if (...) NULL`.
ℹ In group 1: `Group = "Age: 17-19"`.
Caused by warning:
! `cur_data_all()` was deprecated in dplyr 1.1.0.
ℹ Please use `pick()` instead.

Summary tables saved to: mc_cfa_summary_tables_SSQ-10_(Theoretical_Depressive_Symptoms)_ka_k1_n1000.rds


Generating distribution plots...

ℹ Please use `linewidth` instead.

Figures saved to: /Users/Marothi.Letsoalo/Library/CloudStorage/OneDrive-SharedLibraries-AHRI/Data Science Unit - Biostatistics/team_members/Marothi/projects/CO-LUMINATE/OBJ03-Psychometric/statistical_analysis/output/figures

Plot objects saved to: mc_cfa_plot_objects_SSQ-10_(Theoretical_Depressive_Symptoms)_ka_k1_n1000.rds

Data loaded successfully. Rows: 843 Cols: 44

Target Set: SSQ-10 (Theoretical Depressive Symptoms)

Items: SSQ01, SSQ02, SSQ03, SSQ08, SSQ09, SSQ10, SSQ11, SSQ12, SSQ13, SSQ14

Manual Override: Parallel with 3 Factors

Warning in select_mc_seeds(valid_runs = valid_runs, winning_sig = winning_sig,
: No seeds matched the winning signature; returning empty vector.

Re-using existing Monte Carlo results from disk.

Summary tables saved to: mc_cfa_summary_tables_SSQ-10_(Theoretical_Depressive_Symptoms)_pa_k3_n1000.rds


Generating distribution plots...

Figures saved to: /Users/Marothi.Letsoalo/Library/CloudStorage/OneDrive-SharedLibraries-AHRI/Data Science Unit - Biostatistics/team_members/Marothi/projects/CO-LUMINATE/OBJ03-Psychometric/statistical_analysis/output/figures

Plot objects saved to: mc_cfa_plot_objects_SSQ-10_(Theoretical_Depressive_Symptoms)_pa_k3_n1000.rds

Data loaded successfully. Rows: 843 Cols: 44

Target Set: PHQ-09 (Original)

Items: PHQ901, PHQ902, PHQ903, PHQ904, PHQ905, PHQ906, PHQ907, PHQ908, PHQ909

Initial Consensus: Kaiser suggested 1 Factors

Warning in select_mc_seeds(valid_runs = valid_runs, winning_sig = winning_sig,
: No seeds matched the winning signature; returning empty vector.

Re-using existing Monte Carlo results from disk.

Summary tables saved to: mc_cfa_summary_tables_PHQ-09_(Original)_ka_k1_n1000.rds


Generating distribution plots...

Figures saved to: /Users/Marothi.Letsoalo/Library/CloudStorage/OneDrive-SharedLibraries-AHRI/Data Science Unit - Biostatistics/team_members/Marothi/projects/CO-LUMINATE/OBJ03-Psychometric/statistical_analysis/output/figures

Plot objects saved to: mc_cfa_plot_objects_PHQ-09_(Original)_ka_k1_n1000.rds

Data loaded successfully. Rows: 843 Cols: 44

Target Set: PHQ-09 (Original)

Items: PHQ901, PHQ902, PHQ903, PHQ904, PHQ905, PHQ906, PHQ907, PHQ908, PHQ909

Manual Override: Parallel with 3 Factors

Warning in select_mc_seeds(valid_runs = valid_runs, winning_sig = winning_sig,
: No seeds matched the winning signature; returning empty vector.

Re-using existing Monte Carlo results from disk.

Summary tables saved to: mc_cfa_summary_tables_PHQ-09_(Original)_pa_k3_n1000.rds


Generating distribution plots...

Figures saved to: /Users/Marothi.Letsoalo/Library/CloudStorage/OneDrive-SharedLibraries-AHRI/Data Science Unit - Biostatistics/team_members/Marothi/projects/CO-LUMINATE/OBJ03-Psychometric/statistical_analysis/output/figures

Plot objects saved to: mc_cfa_plot_objects_PHQ-09_(Original)_pa_k3_n1000.rds

## Introduction

The primary aim of this study is to establish a rigorous, evidence-based diagnostic cut-off score for the Shona Symptom Questionnaire (SSQ), specifically the theoretically derived 10-item subset (SSQ-10), using the Patient Health Questionnaire-9 (PHQ-9) as a reference standard for probable depression. To achieve this aim, the study first evaluates the structural validity and stability of the SSQ by comparing the original 14-item scale against the theoretical SSQ-10 and the existing 8-item Dixon-Chibanda subset ([Chibanda et al. 2019](#ref-chibanda2019ssq8)) to identify the optimal configuration for clinical use. Subsequently, the analysis validates measurement invariance to confirm that the SSQ-10 functions consistently across depression status, biological sex, and age groups, ensuring the derived cut-off is universally applicable without bias ([Putnick and Bornstein 2016](#ref-putnick2016measurement)). Finally, the study derives a harmonized diagnostic threshold to facilitate accurate depression screening and comparability between the SSQ-10 and the PHQ-9 among adolescents and young adults.

## Methods

**Setting and Study Design**

This study is embedded within the Isisekelo Sempilo (“Foundation of Life”) trial, a $2 \times 2$ factorial randomized controlled trial conducted by the Africa Health Research Institute (AHRI) in the uMkhanyakude district of rural KwaZulu-Natal, South Africa ([Chidumwa et al. 2022](#ref-chidumwa2022isisekelo)). The trial aimed to evaluate interventions integrating HIV prevention and care within sexual and reproductive health services. The study setting is predominantly rural and characterized by high levels of poverty and unemployment, with over 85% of young adults aged 20–24 reported as unemployed. The AHRI demographic surveillance area covers a population of approximately 140,000, characterized by a high prevalence of HIV.

**Participants and Sampling**

The trial utilized the AHRI demographic surveillance system as a sampling frame to randomly select 3,000 men and women aged 16–29 years, stratified by sex. Eligibility criteria for the main trial included residence within the surveillance area, willingness to provide informed consent, and willingness to participate in follow-up assessments including HIV testing via dried blood spot.

For the present psychometric analysis, a specific subsample was selected based on age, restricting the cohort to participants aged 16–24 years, and data completeness, requiring concurrent baseline measurements for both the SSQ-14 and the PHQ-9. This selection ensures that the validation and harmonization analyses are conducted on a directly comparable cohort relevant to adolescent and young adult mental health.

**Data Collection Procedures**

Participants were invited to participate via home visits conducted by trained fieldworkers. Upon contact, the study purpose was explained, and informed consent was obtained (written consent for adults; assent plus parental/guardian consent for participants aged 16–17).

Participants completed a structured baseline questionnaire administered electronically by fieldworkers using secure tablets. The interview included sociodemographic details, health care utilization history, and the full SSQ-14 and PHQ-9 instruments. To minimize social desirability bias and ensure comprehension, interviews were conducted in the local language, isiZulu, in private settings within the participant’s household or a preferred community location. Clinical attendance data were captured electronically via the ClinicLink system across 11 clinics within the surveillance area.

**Measures**

*The Patient Health Questionnaire (PHQ-9)*

The PHQ-9 is the depression module of the PRIME-MD diagnostic instrument, designed to align strictly with the criteria for Major Depressive Disorder (MDD) in the DSM-IV ([Kroenke, Spitzer, and Williams 2001](#ref-Kroenke2001)). It consists of 9 items scored on a 4-point Likert scale ranging from “Not at all” (0) to “Nearly every day” (3). It measures the severity of MDD symptoms over the past 2 weeks, covering domains such as anhedonia, depressed mood, sleep disturbance, fatigue, and suicidal ideation. While validated in South Africa, its etic approach may miss non-DSM manifestations of distress common in African contexts.

*The Shona Symptom Questionnaire (SSQ-14)*

The SSQ-14 represents an emic alternative. Developed in Zimbabwe, it was derived from ethnographic research into how Shona speakers articulate mental distress ([Patel et al. 1997](#ref-Patel1997)). It consists of 14 items with a binary (Yes/No) response format and measures Common Mental Disorders (CMD)—a blend of depression and anxiety—over the past 7 days. It includes indigenous idioms such as kufungisisa (“thinking too much”) and somatic sensations. While culturally resonant, its binary format and mixing of anxiety and depression items present specific challenges for harmonisation with the polytomous, depression-specific PHQ-9 ([Haney et al. 2014](#ref-Haney2014)).

*Derivation of the SSQ-10 (Theoretical Subset)*

To enhance the specificity of the SSQ for depression screening, a theoretically derived subset (SSQ-10) was configured by retaining items that conceptually align with DSM-5 and ICD-11 criteria for a depressive episode, while excluding non-specific distress items ([Patel et al. 1997](#ref-Patel1997)).

Ten items were retained: thinking too much (cognitive distress/rumination), difficulty concentrating (cognitive), irritability (affective-arousal), poor sleep (somatic-affective), crying (affective), feeling tired (somatic-affective), suicidal thoughts (cognitive-affective), general unhappiness (affective-cognitive), work lagging behind (functional/cognitive), and difficulty deciding (cognitive). Irritability was specifically retained as it is recognized as an alternative mood expression for youth depression in DSM-5.

Four items were excluded from the theoretical subset due to conceptual divergence from the core depression construct. Nightmares (SSQ04) were removed as they overlap with sleep disturbance but align more closely with anxiety or trauma. Seeing or hearing things (SSQ05) was excluded as it represents perceptual disturbances associated with severe or psychotic depression rather than core depressive criteria. Stomach ache (SSQ06) was excluded to reduce confounding with general somatic complaints common in this setting, while being frightened by trivial things (SSQ07) was removed as it serves as a marker for anxiety or hyperarousal rather than depression.

**Data Analysis**

All statistical analyses were conducted using R Statistical Software ([R Core Team 2023](#ref-rcoreteam2023)).

*Descriptive Statistics*

Descriptive statistics were computed to summarize the demographic and clinical characteristics of the sample. Categorical variables (e.g., biological sex, employment status, depression status) were reported as frequencies and percentages. Continuous variables (e.g., age, PHQ-9 total score) were summarized using means and standard deviations (SD). The distribution of item responses was examined to identify endorsement rates and potential floor or ceiling effects.

*Structural Validity and Stability*

To determine the most robust version of the SSQ for screening purposes, three item configurations were evaluated: the original 14-item scale (SSQ-14), a theoretically derived 10-item subset focusing on core depressive symptoms (SSQ-10), and the 8-item Dixon-Chibanda subset ([Chibanda et al. 2019](#ref-chibanda2019ssq8)).

A Monte Carlo split-sample resampling approach was employed to rigorously test the stability of these structures. The dataset was randomly partitioned into exploratory and confirmatory halves for 10,000 iterations. In each iteration, Exploratory Factor Analysis (EFA) was conducted to identify the most likely factor structure using established criteria appropriate for categorical data, specifically the Kaiser Criterion and Parallel Analysis, utilizing polychoric correlations ([Holgado–Tello et al. 2010](#ref-holgado2010polychoric)). The Consensus Model was then identified as the specific grouping of items that appeared most frequently across these replications, serving as the definitive structure for each scale version.

Following the exploratory phase, the stability of the Consensus Model for the SSQ-10 was tested using Confirmatory Factor Analysis (CFA) on the hold-out data samples using the Weighted Least Squares Mean and Variance adjusted (WLSMV) estimator.

*Reliability and Measurement Invariance*

Internal consistency reliability was calculated using both Cronbach’s Alpha and McDonald’s Omega Total to provide a comprehensive assessment of the scale’s precision.

To ensure the scale performs fairly across different populations, measurement invariance was assessed. This procedure evaluated whether the SSQ-10 measures depression consistently regardless of a participant’s biological sex, age group, or depression severity status ([Putnick and Bornstein 2016](#ref-putnick2016measurement)).

*Diagnostic Cut-offs and Harmonisation*

To drive diagnostic cut-offs and harmonisation, an optimal screening cut-off for the SSQ-10 was established using the PHQ-9 as the clinical reference standard. Two complementary methods were applied. First, Receiver Operating Characteristic (ROC) analysis was used to identify the score threshold that maximized the balance between sensitivity and specificity. Second, Item Response Theory (IRT) was employed to overcome the limitations of simple binary cut-offs by linking the SSQ-10 and PHQ-9 onto a common latent metric of depression severity ([Samejima 1969](#ref-samejima1969estimation)). This allowed for the identification of the precise level of underlying depression corresponding to the PHQ-9 threshold and mapping it directly to the SSQ-10, creating a harmonized standard for classification ([Kolen and Brennan 2014](#ref-kolen2014test)).

## Results

**Participant Characteristics**

The demographic and clinical characteristics of the study sample ($N=843$) are summarized in <a href="#tbl-demographics" class="quarto-xref">Table 1</a>. The cohort included a relatively balanced distribution of biological sex, with 51% males and 49% females. The age distribution was skewed slightly towards young adults, with 57% of participants aged 20–24 years and 43% aged 17–19 years. The majority of participants resided in rural households (68%), while 28% lived in peri-urban areas and 3.1% in urban settings.

Regarding socioeconomic status, 44% of participants were currently enrolled in school, and 44% were not employed. Notably, 23% were classified as students and thus not applicable for standard unemployment definitions. Food insecurity was reported by 27% of the sample, and 21% were recipients of a government grant. In terms of health and risk factors, 11% of participants tested positive for HIV, 20% reported experiencing violence, and 52% reported alcohol use. Based on the PHQ-9 reference standard, the prevalence of probable depression (score $\ge$ 10) in the sample was 4.0% ($n=34$).

Item-level analysis of the SSQ-14 revealed significant variability in symptom endorsement between depressed and non-depressed groups (<a href="#tbl-item-dist" class="quarto-xref">Table 2</a>). Participants classified as depressed by the PHQ-9 endorsed SSQ items at substantially higher rates than non-depressed participants. For example, thinking too much (SSQ01) was endorsed by 74% of the depressed group compared to 26% of the non-depressed group, and general unhappiness (SSQ12) was endorsed by 59% versus 11%.

Point-biserial correlations confirmed strong associations between individual SSQ items and the continuous PHQ-9 total score, with coefficients ranging from 0.190 to 0.405 (all $p < 0.001$). The strongest correlations with the PHQ-9 total score were observed for thinking too much ($r_{pb}=0.405$), generally unhappy ($r_{pb}=0.405$), and difficulty concentrating ($r_{pb}=0.400$). Conversely, items excluded from the theoretical SSQ-10 subset demonstrated weaker correlations with the depression benchmark, including stomach ache ($r_{pb}=0.190$), hallucinations ($r_{pb}=0.258$), and nightmares ($r_{pb}=0.293$), supporting the theoretical decision to refine the scale for depression specificity.

In [ ]:
source(here("statistical_analysis/programs/utils/descriptive/democraphics_and_item_analysis.R"))

Data loaded successfully.

Tables saved to: /Users/Marothi.Letsoalo/Library/CloudStorage/OneDrive-SharedLibraries-AHRI/Data Science Unit - Biostatistics/team_members/Marothi/projects/CO-LUMINATE/OBJ03-Psychometric/statistical_analysis/output/tables

In [ ]:
tables_1_2$table1

In [ ]:
tables_1_2$table2

**Structural Validity and Stability**

We conducted an Exploratory Factor Analysis (EFA) to determine the optimal factor structure, assessing the baseline stability of the established measures (PHQ-9 and SSQ-14) and investigating the theoretical SSQ-10 subset. The structural validity results are visualized in <a href="#fig-factor-retention" class="quarto-xref">Figure 1</a>, which displays the factor retention probabilities and counts across these three primary configurations based on

In [ ]:
cm_ka_phq9<-gsub(
  "Frequency: ","",
  paste(word(gsub("\n\n","|",
                  mc_efa_results[["PHQ-09 (Original)"]][["cooccurrence"]][["Kaiser"]][["plot_stability"]]@labels$caption),1,sep="\\|"))
  )
cm_pa_phq9<-gsub(
  "Frequency: ","",
  paste(word(gsub("\n\n","|",
                  mc_efa_results[["PHQ-09 (Original)"]][["cooccurrence"]][["Parallel"]][["plot_stability"]]@labels$caption),1,sep="\\|"))
  )


cm_ka_ssq14<-gsub(
  "Frequency: ","",
  paste(word(gsub("\n\n","|",
                  mc_efa_results[["SSQ-14 (Original)"]][["cooccurrence"]][["Kaiser"]][["plot_stability"]]@labels$caption),1,sep="\\|"))
  )
cm_pa_ssq14<-gsub(
  "Frequency: ","",
  paste(word(gsub("\n\n","|",
                  mc_efa_results[["SSQ-14 (Original)"]][["cooccurrence"]][["Parallel"]][["plot_stability"]]@labels$caption),1,sep="\\|"))
  )
cm_pa_ssq08<-gsub(
  "Frequency: ","",
  paste(word(gsub("\n\n","|",
                  mc_efa_results[["SSQ-08 (Dixon Chibanda et.al)"]][["cooccurrence"]][["Parallel"]][["plot_stability"]]@labels$caption),1,sep="\\|"))
  )

cm_pa_ssq10<-gsub(
  "Frequency: ","",
  paste(word(gsub("\n\n","|",
                  mc_efa_results[["SSQ-10 (Theoretical Depressive Symptoms)"]][["cooccurrence"]][["Parallel"]][["plot_stability"]]@labels$caption),1,sep="\\|"))
  )



cm_pa_ssq10_2<-(mc_efa_results[["SSQ-10 (Theoretical Depressive Symptoms)"]][["raw_data"]]) |> filter(p_factor_f==2) |> 
  mutate(N=n()) |> 
  group_by(p_factor_1,p_factor_2,N) |> 
  count() |> 
  ungroup() |> 
  filter(max(n)==n) |> 
  mutate(prop=n/N,
         se=sqrt(prop * (1 - prop) / N),
         ci_lower = max(0, prop - 1.96 * se),
         ci_upper = min(1, prop + 1.96 * se),
         stats_str=sprintf(
           "%d/%d (%.1f%%); 95%% CI: (%.1f%%, %.1f%%)",
           n, N, prop * 100, ci_lower * 100, ci_upper * 100
         )
    
  ) |> pull(stats_str)

In [ ]:
retention_p1<-mc_efa_results[["PHQ-09 (Original)"]][["retention_analysis"]][["plot"]]
retention_p2<-mc_efa_results[["SSQ-14 (Original)"]][["retention_analysis"]][["plot"]]
retention_p3<-mc_efa_results[["SSQ-10 (Theoretical Depressive Symptoms)"]][["retention_analysis"]][["plot"]]
#retention_p3<-mc_efa_results[["SSQ-08 (Dixon Chibanda et.al)"]][["retention_analysis"]][["plot"]]

((retention_p1)+(retention_p2)+(retention_p3)) + 
  patchwork::plot_layout(guides = "collect",
                         axes = "collect_y") +
  patchwork::plot_annotation(
    title=NULL,
    tag_level="A",
    tag_prefix = "(",
    tag_suffix = ")"
                  ) &
  
  ggplot2::theme_minimal(base_size = 16,
                         base_family = "Arial") & 
  ggplot2::labs() & 
  ggsci::scale_fill_jco() &
  ggplot2::theme(
    legend.position = "bottom",
    legend.justification = "left",
    plot.caption = ggplot2::element_text(hjust = 0, size = 12, family = "Arial"),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )&
  guides(color = "none", fill = "none")

We first evaluated the structural properties of the established measures to determine their baseline stability in this population.

For the PHQ-9 reference standard (<a href="#fig-factor-retention" class="quarto-xref">Figure 1</a>, Panel A), a methodological divergence was observed. The Kaiser criterion provided robust support for a unidimensional structure, retaining 1 factor in 95.9% of iterations ($n=9586$). This 2-factor solution favored by the Kaiser criterion—which emerged with the consensus model demonstrates high internal stability of 9586/9586 (100.0%); 95% CI: (100.0%, 100.0%). Conversely, Parallel Analysis (PA) consistently suggested a 3-factor solution (probability = 0.900). The resulting 3-factor consensus model (stability: 4685/9000 (52.1%); 95% CI: (51.0%, 53.1%)) partitioned the scale into three distinct domains: core affective symptoms (PHQ901–902), somatic/vegetative symptoms (PHQ903–905), and cognitive/suicidal symptoms (PHQ906–909). Despite this multidimensional potential, the overwhelming support for the 1-factor solution by the Kaiser criterion justifies the continued use of the PHQ-9 total score as a unidimensional benchmark for depression severity in this cohort. `(....suggest additions)`

For the SSQ-14 (<a href="#fig-factor-retention" class="quarto-xref">Figure 1</a>, Panel B), the Kaiser criterion favored a 2-factor solution in 87.5% of iterations (stability: 5415/8753 (61.9%); 95% CI: (60.8%, 62.9%)), demonstrating high internal stability with items consistently partitioning into distinct cognitive-somatic and affective subdomains. Most notably, thinking too much (SSQ01), difficulty concentrating (SSQ02), and irritability (SSQ03) co-occurred with greater than 95% probability, identifying them as a robust core symptom cluster. However, the 3-factor solution suggested by Parallel Analysis (probability = 0.696; stability: 1480/6961 (21.3%); 95% CI: (20.3%, 22.2%)) revealed that the added dimensionality was largely driven by the distinct clustering of somatic and perceptual items (e.g., hallucinations, nightmares) into a separate “noise” factor.

In [ ]:
# A: PHQ-09, Parallel
p1_parallel_phq9 <- make_panel(
  scale_key      = "PHQ-09 (Original)",
  method         = "Parallel",
  caption_prefix = "3-Factor Parallel Concensus Model:",
  legend_position = NULL,
  axes_collect_y  = FALSE,
  drop_guides     = TRUE
)

# B: SSQ-10, Parallel
p1_parallel_ssq10 <- make_panel(
  scale_key      = "SSQ-10 (Theoretical Depressive Symptoms)",
  method         = "Parallel",
  caption_prefix = "2-Factor Kaiser Concensus Model:",
  legend_position = NULL,
  axes_collect_y  = FALSE,
  drop_guides     = TRUE
)

# Final combined figure with A/B tags
(p1_parallel_phq9 / p1_parallel_ssq10) +
  patchwork::plot_layout(guides = "collect", axes = "collect_y")+
  patchwork::plot_annotation(
    title      = NULL,
    # if you're still using custom A "" B "" C "" tagging:
    tag_levels = list(c("(A)", "", "(B)", ""))
  ) &
  ggplot2::theme(
    plot.tag = ggplot2::element_text(face = "bold", size = 16),  # style for A/B/C
    legend.position = "bottom",
    plot.caption = ggplot2::element_text(hjust = 0, size = 12, family = "Arial"),
    panel.grid.major.x = ggplot2::element_blank(),
    panel.grid.minor.x = ggplot2::element_blank()
  )

In [ ]:
# B: SSQ-14, Kaiser
p1_kaiser_ssq14 <- make_panel(
  scale_key      = "SSQ-14 (Original)",
  method         = "Kaiser",
  caption_prefix = "2-Factor Kaiser Concensus Model:",
  legend_position = NULL,
  axes_collect_y  = FALSE,
  drop_guides     = TRUE
)

# C: SSQ-14, Parallel
p2_parralel_ssq4 <- make_panel(
  scale_key      = "SSQ-14 (Original)",
  method         = "Parallel",
  caption_prefix = "3-Factor Parallel Concensus Model:",
  legend_position = NULL,
  axes_collect_y  = FALSE,
  drop_guides     = FALSE
)

# Final combined figure with A/B tags
(p1_kaiser_ssq14 / p2_parralel_ssq4) +
  patchwork::plot_layout(guides = "collect", axes = "collect_y")+
  patchwork::plot_annotation(
    title      = NULL,
    # if you're still using custom A "" B "" C "" tagging:
    tag_levels = list(c("(A)", "", "(B)", ""))
  ) &
  ggplot2::theme(
    plot.tag = ggplot2::element_text(face = "bold", size = 16),  # style for A/B/C
    legend.position = "bottom",
    plot.caption = ggplot2::element_text(hjust = 0, size = 12, family = "Arial"),
    panel.grid.major.x = ggplot2::element_blank(),
    panel.grid.minor.x = ggplot2::element_blank()
  )

**The Theoretical Subset Divergence (SSQ-10)**

The theoretical SSQ-10 subset (<a href="#fig-factor-retention" class="quarto-xref">Figure 1</a>, Panel C) exhibited a similar but pronounced conflict between methods. The Kaiser criterion strongly supported a unidimensional structure, retaining 1 factor in 83.1% of iterations ($n=8311$). Conversely, PA suggested a 3-factor solution in 72.2% of iterations ($n=7218$). This robust statistical disagreement—with one method suggesting a single score and the other suggesting three distinct sub-scales—necessitated a deeper investigation to determine whether the third factor represented a substantive latent construct or a statistical artifact.

**Artifact Investigation and Resolution**

To adjudicate the conflicting retention suggestions for the SSQ-10, we examined the clustering of items within the 3-factor solution suggested by Parallel Analysis (consensus stability: 2891/7218 (40.1%); 95% CI: (38.9%, 41.2%)). As illustrated in <a href="#fig-heatmap-ssq10-pa" class="quarto-xref">Figure 4</a> (Panel A), the factor loading pattern revealed a distinct anomaly: while two stable factors emerged representing cognitive and affective symptoms (Factors 1 and 2), the third dimension was revealed to be a “singleton” factor comprised solely of poor sleep (SSQ08). This item was separated from both the cognitive (SSQ01–SSQ03) and general affective (SSQ09–SSQ14) clusters. However, closer inspection of the co-occurrence probabilities reveals that poor sleep (SSQ08) co-occurs significantly more often with the affective items (Factor 1: SSQ09–SSQ14) than with the cognitive items (Factor 2: SSQ01–SSQ03). Specifically, Factor 3 exhibited an average co-occurrence of 29.4% with Factor 1 items compared to only 0.3% with Factor 2 items. This pattern suggests that while statistically isolated in the 3-factor solution, the item shares more residual variance with the affective domain, and the isolation is likely driven by item-specific variance or “noise” rather than a substantive multidimensional trait. Given that a single-item factor lacks theoretical parsimony, we evaluated the stability of the 2-factor solution as the next most plausible structure. Although the 2-factor solution was suggested as the primary structure in only 9% ($899/10,000$) of the initial Parallel Analysis iterations, deeper investigation revealed it to be highly robust when statistical artifacts were resolved. Specifically, the consensus stability for this 2-factor configuration was 846/899 (94.1%); 95% CI: (92.6%, 95.6%). This finding confirms that once the artifactual singleton factor is discounted, the SSQ-10 stabilizes into two robust dimensions: a cognitive-functional factor and an affective-emotional factor.

In [ ]:
set="SSQ-10 (Theoretical Depressive Symptoms)"
cooccurrence_p <- mc_efa_results[[set]][["cooccurrence"]][["Parallel"]][["plot_heatmap"]]
stability_p    <- mc_efa_results[[set]][["cooccurrence"]][["Parallel"]][["plot_stability"]]

# Build caption from the existing caption in stability_p
cap_raw       <- gsub("\n\n", "\n", c(stability_p@labels$caption))
cap_extracted <- paste(word(set,1,sep=" "),"3-Factor Parallel Concensus Model:", cap_raw)

((stability_p&ggsci::scale_fill_jama()&ggplot2::labs(caption = NULL))+(cooccurrence_p)) +
  patchwork::plot_layout(widths = c(1,1),guides = "collect", axes = "collect_y")+
  patchwork::plot_annotation(
    caption = cap_extracted,
    tag_levels = "A"
  ) &
  ggplot2::labs(
    title      = NULL,
    subtitle      = NULL)&
  ggplot2::theme(
    plot.tag = ggplot2::element_text(face = "bold", size = 16),  # style for A/B/C
    legend.position = "bottom",
    plot.caption = ggplot2::element_text(hjust = 0, size = 12, family = "Arial"),
    panel.grid.major.x = ggplot2::element_blank(),
    panel.grid.minor.x = ggplot2::element_blank()
  )

**Other SSQ-14 subset (SSQ-08)**

In addition, we also examined the SSQ-08, a brief subset originally developed and validated for screening Common Mental Disorders (CMD) in populations with high HIV prevalence ([Chibanda et al. 2019](#ref-chibanda2019ssq8)), a context that aligns closely with our study setting. Similar to the other versions, the Kaiser criterion was unequivocal, suggesting a unidimensional (1-factor) structure in 100% of iterations (Supplementary <a href="#fig-ssq08-validity" class="quarto-xref">Figure 5</a>, Panel A). In contrast, PA continued to suggest multidimensionality, retaining 3 factors in 66.3% of the Monte Carlo samples.

Critically, as illustrated in the membership and heatmap plots in Supplementary <a href="#fig-ssq08-validity" class="quarto-xref">Figure 5</a> (Panels B & C), while the existing SSQ-08 subset retains thinking too much (SSQ01), it excludes the highly stable partners of this core cluster, specifically SSQ02 and SSQ03. The omission of these clinically coherent neurocognitive and affective markers, despite their strong structural cohesion with the primary idiom of distress (kufungisisa), suggests a potential limitation in capturing the full breadth of depressive symptomatology in this specific adolescent cohort. This finding explicitly validates the configuration of the SSQ-10, which retains this entire stable core cluster while systematically removing only the unstable ‘noise’ items (e.g., SSQ04–07) that drove the artifactual dimensionality in the original scale.

In [ ]:
set="SSQ-08 (Dixon Chibanda et.al)"
cooccurrence_p <- mc_efa_results[[set]][["cooccurrence"]][["Parallel"]][["plot_heatmap"]]
stability_p    <- mc_efa_results[[set]][["cooccurrence"]][["Parallel"]][["plot_stability"]]
retention_p1<-mc_efa_results[[set]][["retention_analysis"]][["plot"]]+ 
  ggsci::scale_fill_jco()
# Build caption from the existing caption in stability_p
cap_raw       <- gsub("\n\n", "\n", c(stability_p@labels$caption))
cap_extracted <- paste(word(set,1,sep=" "),"3-Factor Parallel Concensus Model:", cap_raw)

((retention_p1)+(stability_p&ggsci::scale_fill_jama()&ggplot2::labs(caption = NULL))+(cooccurrence_p)) +
  patchwork::plot_layout(widths = c(1,1,1),guides = "collect", axes = "collect_y")+
  patchwork::plot_annotation(
    caption = cap_extracted,
    tag_levels = "A"
  ) &
  ggplot2::labs(
    title      = NULL,
    subtitle      = NULL)&
  ggplot2::theme(
    plot.tag = ggplot2::element_text(face = "bold", size = 16),  # style for A/B/C
    legend.position = "bottom",
    plot.caption = ggplot2::element_text(hjust = 0, size = 12, family = "Arial"),
    panel.grid.major.x = ggplot2::element_blank(),
    panel.grid.minor.x = ggplot2::element_blank()
  )

**Confirmatory Validation and Model Comparison**

Following the discovery phase, we subjected the identified structures to rigorous testing using Confirmatory Factor Analysis (CFA) with the Weighted Least Squares Mean and Variance adjusted (WLSMV) estimator.

*Model Fit and Scoring Justification*

We evaluated the stability and fit of the 1-Factor and 2-Factor models for both the SSQ-10 and the PHQ-9 reference standard across 10,000 Monte Carlo samples. <a href="#tbl-model-comp" class="quarto-xref">Table 3</a> presents the global fit indices, internal consistency reliability, and latent factor correlations.

For the PHQ-9, while the 1-Factor model showed excellent CFI (0.994), the RMSEA was marginal (0.076), with only 64.8% of samples meeting the $\le$ 0.08 threshold. The 2-Factor model improved the RMSEA to 0.049. However, the high correlation between the two factors ($\phi = 0.872$) and the overlapping reliability estimates ($\alpha$ & $\omega$ \> .90 for both total and subscales, with 95% CIs consistently excluding values below .90) support the retention of the unidimensional scoring convention.

For the SSQ-10, the contrast was more pronounced. The 1-Factor model demonstrated a poor RMSEA of 0.091 (95% CI: 0.075, 0.107), with only 7.6% of replications indicating acceptable fit. In contrast, the 2-Factor consensus model achieved a near-perfect fit (CFI = 1.000; RMSEA = 0.010), with 100% of samples meeting adequacy thresholds. Despite this structural distinctness, the moderate-to-strong factor correlation ($\phi = 0.792$) and high internal consistency for both the global scale ($\alpha$ & $\omega$ \> .90; 95% CIs \> .96) and subscales (all 95% CI lower bounds \> .93) suggest that the total score remains a meaningful metric for general depression screening, capturing a cohesive underlying construct despite the multidimensional nuances.

In [ ]:
# Helper to build CFA stability tables for a 1-factor vs 2-factor solution
build_cfa_stability_table <- function(results_1factor, results_2factor) {
  
  # Fit indices -------------------------------------------------------------
  fit_tbl <-
    dplyr::bind_rows(
      results_1factor[["gt_tables"]][["gt_fit_source"]] |>
        dplyr::mutate(factor = "1-Factor"),
      results_2factor[["gt_tables"]][["gt_fit_source"]] |>
        dplyr::mutate(factor = "2-Factor")
    ) |>
    dplyr::mutate(
      Prop_Good_CFI   = sprintf("%1.1f", Prop_Good_CFI * 100),
      Prop_Good_RMSEA = sprintf("%1.1f", Prop_Good_RMSEA * 100)
    ) |>
    dplyr::select(
      factor,
      `CFI (95% CI)`, Prop_Good_CFI,
      `RMSEA (95% CI)`, Prop_Good_RMSEA
    )
  
  # Factor correlations (only for 2-factor solution) -----------------------
  factor_corr_tbl <-
    results_2factor[["gt_tables"]][["gt_factor_correlations_source"]] |>
    dplyr::transmute(
      `Factors Corr (95% CI)` = `Corr (95% CI)`,
      factor = "2-Factor"
    )
  
  # Reliability: collapse scales per solution ------------------------------
  reliability_tbl <-
    dplyr::bind_rows(
      results_2factor[["gt_tables"]][["gt_reliability_source"]] |>
        dplyr::mutate(factor = "2-Factor"),
      results_1factor[["gt_tables"]][["gt_reliability_source"]] |>
        dplyr::mutate(factor = "1-Factor")
    ) |>
    dplyr::select(factor, Scale, `Alpha (95% CI)`, `Omega (95% CI)`) |>
    dplyr::group_by(factor) |>
    dplyr::summarise(
      `Alpha (95% CI)` = paste0(Scale, ": ", `Alpha (95% CI)`, collapse = "\n"),
      `Omega (95% CI)` = paste0(Scale, ": ", `Omega (95% CI)`, collapse = "\n"),
      .groups = "drop"
    )
  
  # Combine, reshape, and create row groups --------------------------------
  combined <- fit_tbl |>
    dplyr::left_join(factor_corr_tbl, by = "factor") |>
    dplyr::left_join(reliability_tbl, by = "factor") |>
    dplyr::rename(
      Solution         = factor,
      `% CFI ≥ 0.90`   = Prop_Good_CFI,
      `% RMSEA ≤ 0.08` = Prop_Good_RMSEA
    )
  
  combined_long <- combined |>
    tidyr::pivot_longer(
      cols      = -Solution,
      names_to  = "Measure",
      values_to = "Statistic"
    ) |>
    dplyr::mutate(
      Group = dplyr::case_when(
        Measure %in% c("CFI (95% CI)", "% CFI ≥ 0.90",
                       "RMSEA (95% CI)", "% RMSEA ≤ 0.08") ~ "Fit indices",
        Measure %in% c("Alpha (95% CI)", "Omega (95% CI)") ~ "Reliability",
        Measure %in% c("Factors Corr (95% CI)") ~ "Factor correlations",
        TRUE ~ "Other"
      ),
      Measure = factor(
        Measure,
        levels = c(
          "CFI (95% CI)", "% CFI ≥ 0.90",
          "RMSEA (95% CI)", "% RMSEA ≤ 0.08",
          "Alpha (95% CI)", "Omega (95% CI)",
          "Factors Corr (95% CI)"
        )
      ),
      Group = factor(
        Group,
        levels = c("Fit indices", "Reliability", "Factor correlations", "Other")
      )
    ) |>
    dplyr::arrange(Group, Measure)
  
  combined_long |>
    tidyr::pivot_wider(
      names_from  = Solution,
      values_from = Statistic
    ) |>
    dplyr::mutate(
      Measure = as.character(Measure),
      Group   = as.character(Group)
    ) |>
    dplyr::select(Group, Measure, dplyr::everything())
}

# Build tables for PHQ-9 and SSQ-10 ----------------------------------------

table_phq09_cfa_stability <- build_cfa_stability_table(
  results_1factor = batch_res$phq09_data_driven,
  results_2factor = batch_res$phq09_pa_3f
)

# Remove first columns (Group/Measure) from SSQ-10 table before binding ----

table_ssq10_cfa_stability <- build_cfa_stability_table(
  results_1factor = batch_res$ssq10_data_driven,
  results_2factor = batch_res$ssq10_pa_3f
) |>
  dplyr::select(-Group, -Measure)

# Clean factor column names and prepare for column groups -------------------

table_phq09_cfa_stability <- table_phq09_cfa_stability |>
  dplyr::rename(
    phq9_1factor = `1-Factor`,
    phq9_2factor = `2-Factor`
  )

table_ssq10_cfa_stability <- table_ssq10_cfa_stability |>
  dplyr::rename(
    ssq10_1factor = `1-Factor`,
    ssq10_2factor = `2-Factor`
  )

combined_tbl <- table_phq09_cfa_stability |>
  dplyr::bind_cols(table_ssq10_cfa_stability)

# Final GT table with column and row groups --------------------------------

combined_tbl |>
  gt::gt(groupname_col = "Group") |>
  gt::sub_missing(columns = dplyr::everything(), missing_text = "-") |>
  # Center all non-group columns
  gt::cols_align(
    align   = "center",
    columns = -c(Group, Measure)
  ) |>
  # Clean column labels
  gt::cols_label(
    Measure       = "Measure",
    phq9_1factor  = "1-Factor",
    phq9_2factor  = "2-Factor",
    ssq10_1factor = "1-Factor",
    ssq10_2factor = "2-Factor"
  ) |>
  # Column groups (spanners)
  gt::tab_spanner(
    label   = "PHQ-9",
    columns = c(phq9_1factor, phq9_2factor)
  ) |>
  gt::tab_spanner(
    label   = "SSQ-10",
    columns = c(ssq10_1factor, ssq10_2factor)
  ) |>
  # Bold row groups and column groups
  gt::tab_style(
    style = gt::cell_text(weight = "bold"),
    locations = gt::cells_row_groups()
  ) |>
  gt::tab_style(
    style = gt::cell_text(weight = "bold", align = "center"),
    locations = gt::cells_column_spanners()
  ) |>
  gt::tab_header(
    title    = "Model Comparison (1-Factor vs 2-Factor)",
    subtitle = paste(
      c(
        "Across Monte Carlo Samples: Global Fit Indices",
        "Latent Factor Correlations",
        "Internal Consistency Reliability"
      ),
      collapse = "; "
    )
  ) |>
  gt::tab_source_note(
    source_note = paste(
      c(
        "Note:",
        "CFI/TLI ≥ 0.90 (often ≥ 0.95 for good fit) and RMSEA ≤ 0.08 (≤ 0.06 for good fit) indicate acceptable model fit.",
        "Percentage correspond to the fraction of Monte Carlo samples meeting these benchmarks.",
        "Latent factor correlations estimated from CFA models across Monte Carlo splits, reported as mean and 95% Monte Carlo quantile interval.",
        "Interpretation guide: α, ω ≥ 0.70 acceptable, ≥ 0.80 good, ≥ 0.90 excellent.",
        "Estimates are averaged across Monte Carlo splits with 95% quantile intervals."
      ),
      collapse = "\n"
    )
  )

*Measurement Invariance*

To ensure the equitable utility of the SSQ-10 across different population subgroups, we conducted Multi-Group CFA (MGCFA) using 1,000 Monte Carlo resampling iterations. We tested for measurement invariance across biological sex (Male vs. Female) and age groups (16–19 vs. 20–24). The results for the SSQ-10 (1-Factor & 2-Factor) and PHQ-9 (1-Factor & 2-Factor) models are summarized in Supplementary Table S1.

For the PHQ-9 1-Factor model, the analysis demonstrated robust invariance based on the Comparative Fit Index. Across age groups, 99.7% of Monte Carlo samples met the change in CFI criterion ($\le$ 0.010), and across biological sex, 99.5% met this threshold. However, the Root Mean Square Error of Approximation proved highly unstable, with only 16.8% of age group samples and 44.4% of sex samples meeting the invariance criterion ($\le$ 0.015).

Similarly, the PHQ-9 2-Factor model showed perfect stability regarding the Comparative Fit Index, with 100% of samples meeting the criterion across both age and sex groupings. Yet, the Root Mean Square Error of Approximation remained inconsistent, satisfying the threshold in only 20.3% of age group samples and 46.4% of sex samples. Despite the divergence in the error of approximation metric, the high stability of the Comparative Fit Index supports the conclusion that the PHQ-9 constructs are measured comparably across these groups, as the Comparative Fit Index is often considered a more reliable indicator of invariance in this context.

For the SSQ-10 1-Factor Model, the analysis provided strong evidence for scalar invariance. Across Age Groups, the mean $\Delta$CFI was 0.000, with 96.7% of samples meeting the invariance threshold, and 87.5% meeting the RMSEA criterion. Similarly, across Biological Sex, 95.2% of samples satisfied the $\Delta$CFI cutoff, and 94.9% met the $\Delta$RMSEA criterion. These results indicate that the unidimensional structure of the SSQ-10 is measurement invariant across these demographic factors.

For the SSQ-10 2-Factor Model, the analysis also provided strong support for scalar invariance across Biological Sex. The mean change in CFI was 0.000, with 99.8% of Monte Carlo samples meeting the strict invariance threshold. Root Mean Square Error of Approximation stability was also high, with 89.5% of samples meeting the criterion. For Age Groups, the Comparative Fit Index criterion remained highly robust, with 99.7% of samples meeting the $\le$ 0.010 threshold. However, the Root Mean Square Error of Approximation metric was less stable, with only 56.4% of samples meeting the strict $\le$ 0.015 cutoff. This divergence likely reflects the known sensitivity of the Root Mean Square Error of Approximation to sample size and model complexity in subgroup partitions. Given the near-perfect stability of the Comparative Fit Index, these findings suggest that the SSQ-10 measures the underlying construct of depression in a comparable manner across these key demographic groups, satisfying the prerequisites for valid group comparisons ([Putnick and Bornstein 2016](#ref-putnick2016measurement)).

#### 1-Factor PHQ-9

In [ ]:
batch_res[["phq09_data_driven"]][["gt_tables"]][["invariance"]]

#### 2-Factor PHQ-9

In [ ]:
batch_res[["phq09_pa_3f"]][["gt_tables"]][["invariance"]]

#### 1-Factor SSQ-10

In [ ]:
batch_res[["ssq10_data_driven"]][["gt_tables"]][["invariance"]]

#### 2-Factor SSQ-10

In [ ]:
batch_res[["ssq10_pa_3f"]][["gt_tables"]][["invariance"]]

**Derivation of Diagnostic Cut-offs**

In [ ]:
batch_res[["ssq10_data_driven"]][["gt_tables"]][["roc"]]

In [ ]:
#### SSQ-10 and PHQ-09 (Joint structure)
retention_p<-mc_efa_results[["SSQ-10 and PHQ-09 (Joint Structure)"]][["retention_analysis"]][["plot"]]
retention_p

cooccurrence_p<-mc_efa_results[["SSQ-10 and PHQ-09 (Joint Structure)"]][["cooccurrence"]][["Kaiser"]][["plot_heatmap"]]
stability_p<-mc_efa_results[["SSQ-10 and PHQ-09 (Joint Structure)"]][["cooccurrence"]][["Kaiser"]][["plot_stability"]]

((cooccurrence_p)+(stability_p)) + 
  patchwork::plot_layout() +
  patchwork::plot_annotation(title="Factor Retention and Item co-occurrence for the winning factor count and item stability sorted by Winning Model",
                         tag_level="A"
                  ) &
  ggplot2::theme_minimal(base_size = 16) & 
  ggplot2::theme(legend.position = 'bottom',
                 axis.text.x  = ggplot2::element_text(angle = 45,vjust=.85,hjust=1),
                 )

cooccurrence_p<-mc_efa_results[["SSQ-10 and PHQ-09 (Joint Structure)"]][["cooccurrence"]][["Parallel"]][["plot_heatmap"]]
stability_p<-mc_efa_results[["SSQ-10 and PHQ-09 (Joint Structure)"]][["cooccurrence"]][["Parallel"]][["plot_stability"]]

((cooccurrence_p)+(stability_p)) + 
  patchwork::plot_layout() +
  patchwork::plot_annotation(title="Factor Retention and Item co-occurrence for the winning factor count and item stability sorted by Winning Model",
                         tag_level="A"
                  ) &
  ggplot2::theme_minimal(base_size = 16) & 
  ggplot2::theme(legend.position = 'bottom',
                 axis.text.x  = ggplot2::element_text(angle = 45,vjust=.85,hjust=1),
                 )

## Discussion

\[Discussion will be inserted here\]

## Conclusions

\[Conclusions will be inserted here\]

## Tables

\[Tables will be inserted here\]

## Figure Captions

\[Figure captions will be inserted here\]

**Acknowledgements: **

\[Insert Acknowledgements here\]

**Financial Support**

\[Insert Funding Information here\]

**Conflict of Interest**

> **Data Availability Statement**
>
> To ensure compliance with AHRI’s Data Sharing Policy and Standard Operating Procedures (SOP), researchers requiring access to the datasets must submit a formal request via the AHRI Data Science Unit (DSU).

## References

Chibanda, Dixon et al. 2019. “Validation of the 8-Item Shona Symptom Questionnaire (SSQ-8) in Zimbabwe.” In *AIDSImpact 2019 Conference*. <https://www.aidsimpact.com/abstracts/-KoQrAipsu5F2bWuNDBA>.

Chidumwa, Glory, Natsayi Chimbindi, Carina Herbst, Nonhlanhla Okeselo, Jaco Dreyer, Thembelihle Zuma, Theresa Smith, et al. 2022. “Isisekelo Sempilo Study Protocol for the Effectiveness of HIV Prevention Embedded in Sexual Health with or Without Peer Navigator Support (Thetha Nami) to Reduce Prevalence of Transmissible HIV Amongst Adolescents and Young Adults in Rural KwaZulu-Natal: A 2$\times$ 2 Factorial Randomised Controlled Trial.” *BMC Public Health* 22 (1): 454. <https://doi.org/10.1186/s12889-022-12798-y>.

Haney, E., K. Singh, C. Nyamukapa, et al. 2014. “One Size Does Not Fit All: Psychometric Properties of the Shona Symptom Questionnaire (SSQ) Among Adolescents and Young Adults in Zimbabwe.” *Journal of Affective Disorders* 167: 358–67. <https://doi.org/10.1016/j.jad.2014.06.015>.

Holgado–Tello, Francisco Pablo, Salvador Chac’on–Moscoso, Isabel Barbero–Garc’ ia, and Enrique Vila–Abad. 2010. “Polychoric Versus Pearson Correlations in Exploratory and Confirmatory Factor Analysis of Ordinal Variables.” *Quality & Quantity* 44: 153–66. <https://doi.org/10.1007/s11135-008-9190-y>.

Kolen, Michael J, and Robert L Brennan. 2014. *Test Equating, Scaling, and Linking: Methods and Practices*. 3rd ed. Springer. <https://doi.org/10.1007/978-1-4939-0317-7>.

Kroenke, K., R. L. Spitzer, and J. B. Williams. 2001. “The PHQ-9: Validity of a Brief Depression Severity Measure.” *Journal of General Internal Medicine* 16 (9): 606–13. <https://doi.org/10.1046/j.1525-1497.2001.016009606.x>.

Patel, V., E. Simunyu, F. Gwanzura, G. Lewis, and A. Mann. 1997. “The Shona Symptom Questionnaire: The Development of an Indigenous Measure of Common Mental Disorders in Harare.” *Acta Psychiatrica Scandinavica* 95 (6): 469–75. <https://doi.org/10.1111/j.1600-0447.1997.tb10134.x>.

Putnick, Diane L, and Marc H Bornstein. 2016. “Measurement Invariance Conventions and Reporting: The State of the Art and Future Directions for Psychological Research.” *Developmental Review* 41: 71–90. <https://doi.org/10.1016/j.dr.2016.06.004>.

R Core Team. 2023. *R: A Language and Environment for Statistical Computing*. Vienna, Austria: R Foundation for Statistical Computing.

Samejima, Fumiko. 1969. “Estimation of Latent Ability Using a Response Pattern of Graded Scores.” *Psychometrika Monograph Supplement* 17 (4): 1–100. <https://doi.org/10.1007/BF02290599>.

## Supplementary